addons/account/models/account_move.py

```diff
@@ -1924,6 +1924,33 @@ def _check_journal_type(self):
             if record.is_sale_document() and journal_type != 'sale' or record.is_purchase_document() and journal_type != 'purchase':
                 raise ValidationError(_("The chosen journal has a type that is not compatible with your invoice type. Sales operations should go to 'sale' journals, and purchase operations to 'purchase' ones."))
 
+    @api.constrains('auto_post', 'partner_id', 'invoice_line_ids')
+    def _check_post_entries(self):
+        if not self._context.get('check_auto_post'):
+            self = self.filtered(lambda m: m.auto_post)
```

Comment:
Why do you need this?

Reply:
@william-andre  
It is used for post the move manually which is auto_post = False.